java/vector/src/main/codegen/templates/ComplexCopier.java

```diff
@@ -58,6 +59,27 @@ private static void writeValue(FieldReader reader, FieldWriter writer) {
           writer.endList();
         }
         break;
+      case MAP:
+        if (reader.isSet()) {
+          writer.startList();
```

Comment:
Why not use the methods in `UnionMapWriter`
```
startMap()
startEntry()
...
endEntry()
endMap()
```

Then you wouldn't need to use `setAddVectorAsNullable`

Reply:
thanks for your suggestion, fixed.